status.tmpl

```diff
@@ -162,67 +222,108 @@
           }
         }
       };
+
+      var countNodeStatuses = function() {
+        var dbs = {{ marshal . }};
+        var elements = document.getElementsByClassName('versionstate');
+        [].forEach.call(elements, function (elem) {
+          var dbName = elem.getAttribute('data-dbName');
+          var versionName = elem.getAttribute('data-versionName');
+
+          var statusCounts = {};
+          var nodes = dbs.dbs[dbName].versions[versionName].nodes;
+          Object.values(nodes).forEach(function (node) {
+            if (node.state in statusCounts) {
+              statusCounts[node.state]++
+            } else {
+              statusCounts[node.state] = 1;
+            }
+          });
+
+          for (status in statusCounts) {
+            var count = document.createElement('span');
+            count.innerHTML = statusCounts[status].toString() + 'x';
+            count.classList.add('count')
+            elem.appendChild(count);
+
+            var label = document.createElement('span');
+            label.innerHTML = status;
+            label.classList.add('label');
+            label.classList.add(status);
+            elem.appendChild(label);
+          }
+        });
+      };
     </script>
   </head>
 
-  <body onload="drawMaps();">
+  <body {{ if eq (len $.DBs) 1 }} onload="drawMaps();" {{ else }} onload="countNodeStatuses();" {{ end }}>
     <div id="header">
       <h1>Sequins!</h1>
     </div>
+
     <div id="wrapper">
       {{ range $dbName, $db := .DBs}}
-      <div class="db">
-        <h2 class="dbname">/{{ $dbName }} {{ if gt (len $.DBs) 1 }}<a href="/{{ $dbName }}">&rarr;</a>{{ end }}</h2>
-        {{ range $versionName, $version := $db.Versions }}
-        <div class="version">
-          <span class="versionname">/{{ $versionName }}</span><span class="versionpath">({{ $version.Path }})</span>
-          <div class="versionwrapper">
-            <div class="nodes">
-              {{ range $nodeName, $node := $version.Nodes }}
-              <div class="node {{ if $node.Current }}nodecurrent{{else}}nodenotcurrent{{end}}{{ if eq $node.State "ERROR" }} nodeerror{{end}}">
-                <div class="nodename">{{$nodeName}}</div>
-                <div class="nodestatus">
-                {{ if eq $node.State "BUILDING" }}
-                  (building since {{ $node.CreatedAt }})
-                {{ else if eq $node.State "ERROR" }}
-                  (errored)
-                {{ else }}
-                  (available since {{ $node.AvailableAt }})
-                {{ end }}
+        <div class="db">
+          <h2 class="dbname">/{{ $dbName }} {{ if gt (len $.DBs) 1 }}<a href="/{{ $dbName }}">&rarr;</a>{{ end }}</h2>
+          {{ range $versionName, $version := $db.Versions }}
+
+          <div class="version">
+            <span class="versionname">/{{ $versionName }}</span>
+
+            {{ if gt (len $.DBs) 1 }}
```

Comment:
I should note that lines like these are a hacky result of the fact that sequins uses the same template to serve status pages for all databases and 1 database. I saw it used as a way to render the page a bit differently based on if we're looking at all databases v. just one so I did the same. Definitely not ideal though.

Reply:
Yup, that's pretty odd. As a simple clean-up, we could at least place a boolean on the view model stating if there's just 1 so you don't have to check the length of the list over and over.

Reply:
So I tried looking a way around this and the view model is the sequins struct used in the actual gocode, so I could add a method to it, but it seemed strange to add a method that would be added just for the benefit of this single use case.

Reply:
When utilized to it's fullest, the purpose of the view model is to remove as much logic from the view as possible, moving it to the controller, where it's more readily testable. I wouldn't add a method to the struct so much as a property calculated in the go code.